### PR TITLE
fix flow-isolation diagnostic message crash

### PR DESCRIPTION
**Pull request by kavon**
resolves rdar://88776902

### CODE REVIEW

**Comment by kavon** [comment]
@swift-ci please test and merge

**Comment by kavon** [comment]
@swift-ci please test

File: lib/SILOptimizer/Mandatory/FlowIsolation.cpp [hunk 1]
```diff
@@ -367,6 +367,14 @@ static StringRef verbForInvoking(ValueDecl *value) {
   return "calling ";
 }
 
+template<typename T>
+static DeclName getNameOrBlank(T* nameable) {
+  if (nameable->hasName())
+    return nameable->getName();
+
+  return DeclName(Identifier::getFromOpaquePointer((void*)"_"));
```
[the remark this answers is not in the document]
Comment (kavon) [reply lib/SILOptimizer/Mandatory/FlowIsolation.cpp:375]: I couldn't find that one, but after around I found `ASTContext::getIdentifier(StringRef)` so I'll use that. Thanks 😄

**Comment by kavon** [comment]
@swift-ci please smoke test and merge

File: lib/SILOptimizer/Mandatory/FlowIsolation.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (slavapestov) [reply lib/SILOptimizer/Mandatory/FlowIsolation.cpp]: That's what I meant, thanks!

**Comment by kavon** [comment]
I figured out what went wrong; the decl name was not it.

@swift-ci please smoke test and merge

**Comment by kavon** [comment]
@swift-ci please smoke test and merge

**Pull request merged**